src/all/comicgrowl/src/eu/kanade/tachiyomi/extension/all/comicgrowl/ComicGrowl.kt

```diff
@@ -1,63 +1,330 @@
+            name = element.selectFirst(".series-ep-list-item-h-text")!!.text()
+            date_upload = element.selectFirst(".series-ep-list-date-time")!!.attr("datetime").toDate()
+            scanlator = PUBLISHER
         }
-        return GET(baseUrl, headers) // Currently just get all ongoing works
+    }
+
+    override fun pageListRequest(chapter: SChapter) = GET(chapter.url)
+
+    override fun pageListParse(document: Document): List<Page> {
+        val pageList = mutableListOf<Page>()
+
+        val viewer = document.selectFirst("#comici-viewer")!!
+        val comiciViewerId = viewer.attr("comici-viewer-id")
+        val memberJwt = viewer.attr("data-member-jwt")
+        val requestUrl = baseUrl.toHttpUrl().newBuilder()
+            .addPathSegment("book")
+            .addPathSegments("contentsInfo")
+            .addQueryParameter("comici-viewer-id", comiciViewerId)
+            .addQueryParameter("user-id", memberJwt)
+            .addQueryParameter("page-from", "0")
+
+        // Initial request to get total pages
+        val initialRequest = GET(requestUrl.addQueryParameter("page-to", "1").build())
+        client.newCall(initialRequest).execute().use { initialResponse ->
+            if (initialResponse.code != 200) {
+                throw Exception("Failed to get page list")
+            }
+            // FIXME: use util in core
+            val totalPages =
+                json.parseToJsonElement(initialResponse.body.string()).jsonObject["totalPages"]!!.jsonPrimitive.content
+            // Get all pages
+            val getAllPagesRequest = GET(requestUrl.setQueryParameter("page-to", totalPages).build())
+            client.newCall(getAllPagesRequest).execute().use {
+                if (it.code != 200) {
+                    throw Exception("Failed to get page list")
+                }
+                val result = json.parseToJsonElement(it.body.string())
+                val resultJson = result.jsonObject["result"]!!.jsonArray
+                resultJson.forEach { resultJsonElement ->
+                    val jsonObject = resultJsonElement.jsonObject
+                    // Add fragment to let interceptor to descramble the image
+                    val scramble = jsonObject["scramble"]!!.jsonPrimitive.content.drop(1).dropLast(1).replace(", ", "-")
+                    val imageUrl =
+                        jsonObject["imageUrl"]!!.jsonPrimitive.content.toHttpUrl().newBuilder().fragment(scramble)
+                            .build()
+                    pageList.add(
+                        Page(
+                            index = jsonObject["sort"]!!.jsonPrimitive.int,
+                            imageUrl = imageUrl.toString(),
+                        ),
+                    )
+                }
+            }
+        }
+        return pageList
+    }
+
+    override fun imageRequest(page: Page): Request {
+        val newHeaders = headersBuilder()
+            .set("Referer", baseUrl)
```

Comment:
Browsers format it like:
```suggestion
            .set("Referer", "$baseUrl/")
```